crates/auth/src/api/handlers/mod.rs

```diff
@@ -103,21 +103,24 @@ pub async fn health_handler(state: Extension<Arc<AppState>>) -> impl IntoRespons
 ///
 /// * `impl IntoResponse` - The response
 pub async fn providers_handler(state: Extension<Arc<AppState>>) -> impl IntoResponse {
-    let providers = state
-        .0
-        .auth_service
-        .providers()
-        .iter()
-        .map(|provider| {
-            json!({
-                "name": provider.name(),
-                "type": provider.provider_type(),
-                "description": provider.description(),
-                "configured": provider.is_configured(),
-                "config": provider.get_config_options(),
-            })
-        })
-        .collect::<Vec<_>>();
+    let mut providers = Vec::new();
+
+    // Check each provider's configuration status
+    // Each provider implements its own logic for what "configured" means
+    for provider in state.0.auth_service.providers() {
+        let is_configured = provider
+            .is_configured_with_users()
+            .await
+            .unwrap_or_else(|_| provider.is_configured());
```

Comment:
I think it makes sense to add a `tracing::error` log on `unwrap_or_else()`, so if provider/DB didn't respond, we have a way to know it instead of just having a silent substitution (that might be hard to debug).